src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/InstalledPackageFeed.cs

```diff
@@ -61,6 +62,20 @@ internal async Task<IPackageSearchMetadata[]> GetMetadataForPackagesAndSortAsync
             return items.OrderBy(p => p.Identity.Id).ToArray();
         }
 
+        internal static async Task<IPackageSearchMetadata[]> GetMetadataFromIdentityForPackagesAndSortAsync<T>(T[] packages, bool includePrerelease, CancellationToken cancellationToken) where T : PackageIdentity
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            IEnumerable<IPackageSearchMetadata> items = await TaskCombinators.ThrottledAsync(
+                packages,
+                (p, t) => Task.FromResult(PackageSearchMetadataBuilder.FromIdentity(p).Build()),
+                cancellationToken);
+
+            // The packages were originally sorted which is important because we skip based on that sort
```

Comment:
Question: What does it mean that we skip based on the sort? Maybe this is obvious, but if not, maybe the comment could use a little detail about that so it's clearer when we look at this again in the future.